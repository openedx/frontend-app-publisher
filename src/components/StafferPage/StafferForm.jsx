import React from 'react';
import { Field, FieldArray, reduxForm } from 'redux-form';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';

import ActionButton from '../ActionButton';
import AreasOfExpertise from './AreasOfExpertise';
import SocialLinks from './SocialLinks';
import ImageUpload from '../ImageUpload';
import RenderInputTextField from '../RenderInputTextField';
import ReduxFormCreatableSelect from '../ReduxFormCreatableSelect';
import RichEditor from '../RichEditor';
import FieldLabel from '../FieldLabel';
import ButtonToolbar from '../ButtonToolbar';
import { basicValidate, handleStafferOrCreateFormFail } from '../../utils/validation';

const BaseStafferForm = ({
  handleSubmit,
  pristine,
  isSaving,
  isCreateForm,
  sourceInfo: { referrer },
  cancelStafferInfo,
  organizationName,
  publisherOrganizations,
}) => {
  const formControlDisabled = pristine || isSaving;
  const allOrganizations = organizationName
    ? [...publisherOrganizations.map(org => org.name), organizationName]
    : [...publisherOrganizations.map(org => org.name)];
  const allUniqueOrganizations = [...new Set(allOrganizations)];

  return (
    <div className="create-staffer-form">
      <form onSubmit={handleSubmit}>
        <Field
          name="profile_image.medium.url"
          component={ImageUpload}
          label={(
            <FieldLabel
              id="image.label"
              text="Image"
              helpText={(
                <div>
                  <p>Image Requirements:</p>
                  <ul>
                    <li>The image dimensions must be 110Ã—110.</li>
                    <li>The image size must be less than 256KB.</li>
                  </ul>
                </div>
              )}
              extraText="Image must be 110x110 pixels in size."
            />
          )}
          id="profile_image"
          maxImageSizeKilo={256}
          requiredWidth={110}
          requiredHeight={110}
          className="staffer-image"
          required={isCreateForm}
          validate={basicValidate}
        />
        <Field
          name="given_name"
          component={RenderInputTextField}
          type="text"
          label={<FieldLabel text="First name" />}
          required
        />
        <Field
          name="family_name"
          component={RenderInputTextField}
          type="text"
          label={<FieldLabel text="Last name" />}
          required
        />
        <Field
          name="position.title"
          component={RenderInputTextField}
          type="text"
          label={(
            <FieldLabel
              id="title.label"
              text="Title"
              helpText={(
                <div>
                  <p>Instructor&apos;s title at your organization.</p>
                  <p><b>Examples:</b></p>
                  <ul>
                    <li>Professor</li>
                    <li>Content Developer</li>
                    <li>Director</li>
                  </ul>
                </div>
              )}
            />
          )}
          required
        />
        <Field
          name="position.organization_override"
          component={ReduxFormCreatableSelect}
          label={<FieldLabel text="Organization" />}
          currentValue={organizationName}
          options={
            allUniqueOrganizations.map(org => ({
              label: org,
              value: org,
            }))
          }
          isCreatable
          formatCreateLabel={(label) => label}
          validate={basicValidate}
          placeholder="Organization..."
        />
        <Field
          name="bio"
          component={RichEditor}
          label={<FieldLabel text="Biography" />}
          maxChars={1000}
          validate={basicValidate}
          id="bio"
        />
        <Field
          name="major_works"
          component={RichEditor}
          label={<FieldLabel text="Major works" optional />}
          maxChars={250}
          id="works"
        />
        <FieldLabel text="Social links" className="mb-2" optional />
        <FieldArray
          name="urls_detailed"
          component={SocialLinks}
        />
        <FieldLabel text="Areas of expertise" className="mb-2" optional />
        <FieldArray
          name="areas_of_expertise"
          component={AreasOfExpertise}
        />
        <ButtonToolbar>
          <Link
            className="btn btn-outline-primary form-cancel-btn"
            to={referrer || '/'}
            disabled={formControlDisabled}
            onClick={cancelStafferInfo}
          >
            Cancel
          </Link>
          <ActionButton
            disabled={formControlDisabled}
            labels={isCreateForm ? {
              default: 'Create',
              pending: 'Creating',
            } : {
              default: 'Save',
              pending: 'Saving',
            }}
            state={isSaving ? 'pending' : 'default'}
          />
        </ButtonToolbar>
      </form>
    </div>
  );
};

BaseStafferForm.propTypes = {
  cancelStafferInfo: PropTypes.func,
  handleSubmit: PropTypes.func.isRequired,
  isSaving: PropTypes.bool,
  isCreateForm: PropTypes.bool,
  organizationName: PropTypes.string,
  pristine: PropTypes.bool.isRequired,
  sourceInfo: PropTypes.shape({
    referrer: PropTypes.string,
  }).isRequired,
  publisherOrganizations: PropTypes.arrayOf(PropTypes.shape({
    name: PropTypes.string.isRequired,
  })),
};

BaseStafferForm.defaultProps = {
  cancelStafferInfo: () => {},
  isSaving: false,
  isCreateForm: false,
  organizationName: '',
  publisherOrganizations: [],
};

export default reduxForm({
  form: 'staffer-form',
  onSubmitFail: handleStafferOrCreateFormFail,
})(BaseStafferForm);
export { basicValidate, BaseStafferForm };
