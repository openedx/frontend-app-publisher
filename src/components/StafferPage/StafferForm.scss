.create-staffer-form {
  .staffer-image {
    .uploaded-image {
      max: {
        // These match the 'medium' size in course-discovery's model definition for the image field
        width: 110px;
        height: 110px;
      }
    }
  }

  // react-select specific styles
  .react-select__menu {
    z-index: 5;
  }

  .react-select__control {
    cursor: text;
  }

  .react-select__control.danger {
    border-color: red;
    box-shadow: none;
  }

  .react-select__control--menu-is-open{
    .react-select__single-value{
      color: #D3D3D3
    }
  }
}
