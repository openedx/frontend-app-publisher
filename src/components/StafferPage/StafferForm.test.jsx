import React from 'react';
import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';

import { BaseStafferForm } from './StafferForm';

const stafferInfo = {
  data: {
    uuid: '14d7368b-51dc-42de-9952-0870bcd1f5f1',
    salutation: null,
    given_name: 'staff',
    family_name: 'er',
    bio: '<p><strong>b</strong>io<i>grapizing</i></p>',
    slug: 'staff-er',
    position: {
      title: 'se√±or',
      organization_name: '',
      organization_id: 1,
      organization_override: null,
      organization_marketing_url: null,
    },
    areas_of_expertise: [
      {
        id: 10,
        value: 'Experting',
      },
    ],
    profile_image: {
      medium: {
        height: 110,
        url: 'http://localhost:18381/media/media/people/profile_images/14d7368b-51dc-42de-9952-0870bcd1f5f1-1ffe0424ec10.medium.jpeg',
        width: 110,
      },
    },
    works: [],
    urls: {
      facebook: null,
      twitter: null,
      blog: null,
    },
    urls_detailed: [
      {
        title: 'Github',
        type: 'others',
        id: 10,
        url: 'https://github.com/openedx',
        display_title: 'Github',
      },
    ],
    email: null,
    profile_image_url: 'http://localhost:18381/media/media/people/profile_images/14d7368b-51dc-42de-9952-0870bcd1f5f1-1ffe0424ec10.jpeg',
    major_works: '<p><em>Workings</em></p>',
    published: true,
  },
  error: null,
  isFetching: false,
  isSaving: false,
};

const noReferrerSourceInfo = {
  referrer: null,
};

const sourceInfo = {
  referrer: 'courses/00000000-0000-0000-0000-000000000000',
};

describe('StafferForm', () => {
  it('renders html correctly', () => {
    const component = shallow(<BaseStafferForm
      pristine
      submitting={false}
      isSaving={false}
      handleSubmit={() => {}}
      sourceInfo={noReferrerSourceInfo}
    />);
    expect(shallowToJson(component)).toMatchSnapshot();
  });

  it('renders html correctly when submitting', () => {
    const component = shallow(<BaseStafferForm
      pristine={false}
      submitting
      isSaving={false}
      handleSubmit={() => {}}
      sourceInfo={noReferrerSourceInfo}
    />);
    expect(shallowToJson(component)).toMatchSnapshot();
  });

  it('renders html correctly when creating', () => {
    const component = shallow(<BaseStafferForm
      pristine={false}
      submitting={false}
      isSaving
      handleSubmit={() => {}}
      sourceInfo={noReferrerSourceInfo}
    />);
    expect(shallowToJson(component)).toMatchSnapshot();
  });

  it('renders correctly with staffer info', () => {
    const component = shallow(<BaseStafferForm
      pristine
      submitting={false}
      isSaving={false}
      stafferInfo={stafferInfo}
      handleSubmit={() => {}}
      sourceInfo={noReferrerSourceInfo}
    />);
    expect(shallowToJson(component)).toMatchSnapshot();
  });

  it('renders correctly when sent from the edit course page', () => {
    const component = shallow(<BaseStafferForm
      pristine
      submitting={false}
      isSaving={false}
      handleSubmit={() => {}}
      sourceInfo={sourceInfo}
    />);
    expect(shallowToJson(component)).toMatchSnapshot();
  });
});
