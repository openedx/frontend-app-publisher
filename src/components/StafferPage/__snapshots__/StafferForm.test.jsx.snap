// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StafferForm renders correctly when sent from the edit course page 1`] = `
<div
  className="create-staffer-form"
>
  <form
    onSubmit={[Function]}
  >
    <Field
      className="staffer-image"
      component={[Function]}
      id="profile_image"
      label={
        <FieldLabel
          className=""
          extraText="Image must be 110x110 pixels in size."
          helpText={
            <div>
              <p>
                Image Requirements:
              </p>
              <ul>
                <li>
                  The image dimensions must be 110×110.
                </li>
                <li>
                  The image size must be less than 256KB.
                </li>
              </ul>
            </div>
          }
          id="image.label"
          optional={false}
          text="Image"
        />
      }
      maxImageSizeKilo={256}
      name="profile_image.medium.url"
      required={false}
      requiredHeight={110}
      requiredWidth={110}
      validate={[Function]}
    />
    <Field
      component={[Function]}
      label={
        <FieldLabel
          className=""
          extraText=""
          helpText=""
          id={null}
          optional={false}
          text="First name"
        />
      }
      name="given_name"
      required={true}
      type="text"
    />
    <Field
      component={[Function]}
      label={
        <FieldLabel
          className=""
          extraText=""
          helpText=""
          id={null}
          optional={false}
          text="Last name"
        />
      }
      name="family_name"
      required={true}
      type="text"
    />
    <Field
      component={[Function]}
      label={
        <FieldLabel
          className=""
          extraText=""
          helpText={
            <div>
              <p>
                Instructor's title at your organization.
              </p>
              <p>
                <b>
                  Examples:
                </b>
              </p>
              <ul>
                <li>
                  Professor
                </li>
                <li>
                  Content Developer
                </li>
                <li>
                  Director
                </li>
              </ul>
            </div>
          }
          id="title.label"
          optional={false}
          text="Title"
        />
      }
      name="position.title"
      required={true}
      type="text"
    />
    <Field
      component={[Function]}
      currentValue=""
      formatCreateLabel={[Function]}
      isCreatable={true}
      label={
        <FieldLabel
          className=""
          extraText=""
          helpText=""
          id={null}
          optional={false}
          text="Organization"
        />
      }
      name="position.organization_override"
      options={Array []}
      placeholder="Organization..."
      validate={[Function]}
    />
    <Field
      component={[Function]}
      id="bio"
      label={
        <FieldLabel
          className=""
          extraText=""
          helpText=""
          id={null}
          optional={false}
          text="Biography"
        />
      }
      maxChars={1000}
      name="bio"
      validate={[Function]}
    />
    <Field
      component={[Function]}
      id="works"
      label={
        <FieldLabel
          className=""
          extraText=""
          helpText=""
          id={null}
          optional={true}
          text="Major works"
        />
      }
      maxChars={250}
      name="major_works"
    />
    <FieldLabel
      className="mb-2"
      extraText=""
      helpText=""
      id={null}
      optional={true}
      text="Social links"
    />
    <FieldArray
      component={[Function]}
      name="urls_detailed"
    />
    <FieldLabel
      className="mb-2"
      extraText=""
      helpText=""
      id={null}
      optional={true}
      text="Areas of expertise"
    />
    <FieldArray
      component={[Function]}
      name="areas_of_expertise"
    />
    <ButtonToolbar
      className=""
      leftJustify={false}
    >
      <Link
        className="btn btn-outline-primary form-cancel-btn"
        disabled={true}
        onClick={[Function]}
        to="courses/00000000-0000-0000-0000-000000000000"
      >
        Cancel
      </Link>
      <ActionButton
        className=""
        disabled={true}
        labels={
          Object {
            "default": "Save",
            "pending": "Saving",
          }
        }
        primary={true}
        state="default"
      />
    </ButtonToolbar>
  </form>
</div>
`;

exports[`StafferForm renders correctly with staffer info 1`] = `
<div
  className="create-staffer-form"
>
  <form
    onSubmit={[Function]}
  >
    <Field
      className="staffer-image"
      component={[Function]}
      id="profile_image"
      label={
        <FieldLabel
          className=""
          extraText="Image must be 110x110 pixels in size."
          helpText={
            <div>
              <p>
                Image Requirements:
              </p>
              <ul>
                <li>
                  The image dimensions must be 110×110.
                </li>
                <li>
                  The image size must be less than 256KB.
                </li>
              </ul>
            </div>
          }
          id="image.label"
          optional={false}
          text="Image"
        />
      }
      maxImageSizeKilo={256}
      name="profile_image.medium.url"
      required={false}
      requiredHeight={110}
      requiredWidth={110}
      validate={[Function]}
    />
    <Field
      component={[Function]}
      label={
        <FieldLabel
          className=""
          extraText=""
          helpText=""
          id={null}
          optional={false}
          text="First name"
        />
      }
      name="given_name"
      required={true}
      type="text"
    />
    <Field
      component={[Function]}
      label={
        <FieldLabel
          className=""
          extraText=""
          helpText=""
          id={null}
          optional={false}
          text="Last name"
        />
      }
      name="family_name"
      required={true}
      type="text"
    />
    <Field
      component={[Function]}
      label={
        <FieldLabel
          className=""
          extraText=""
          helpText={
            <div>
              <p>
                Instructor's title at your organization.
              </p>
              <p>
                <b>
                  Examples:
                </b>
              </p>
              <ul>
                <li>
                  Professor
                </li>
                <li>
                  Content Developer
                </li>
                <li>
                  Director
                </li>
              </ul>
            </div>
          }
          id="title.label"
          optional={false}
          text="Title"
        />
      }
      name="position.title"
      required={true}
      type="text"
    />
    <Field
      component={[Function]}
      currentValue=""
      formatCreateLabel={[Function]}
      isCreatable={true}
      label={
        <FieldLabel
          className=""
          extraText=""
          helpText=""
          id={null}
          optional={false}
          text="Organization"
        />
      }
      name="position.organization_override"
      options={Array []}
      placeholder="Organization..."
      validate={[Function]}
    />
    <Field
      component={[Function]}
      id="bio"
      label={
        <FieldLabel
          className=""
          extraText=""
          helpText=""
          id={null}
          optional={false}
          text="Biography"
        />
      }
      maxChars={1000}
      name="bio"
      validate={[Function]}
    />
    <Field
      component={[Function]}
      id="works"
      label={
        <FieldLabel
          className=""
          extraText=""
          helpText=""
          id={null}
          optional={true}
          text="Major works"
        />
      }
      maxChars={250}
      name="major_works"
    />
    <FieldLabel
      className="mb-2"
      extraText=""
      helpText=""
      id={null}
      optional={true}
      text="Social links"
    />
    <FieldArray
      component={[Function]}
      name="urls_detailed"
    />
    <FieldLabel
      className="mb-2"
      extraText=""
      helpText=""
      id={null}
      optional={true}
      text="Areas of expertise"
    />
    <FieldArray
      component={[Function]}
      name="areas_of_expertise"
    />
    <ButtonToolbar
      className=""
      leftJustify={false}
    >
      <Link
        className="btn btn-outline-primary form-cancel-btn"
        disabled={true}
        onClick={[Function]}
        to="/"
      >
        Cancel
      </Link>
      <ActionButton
        className=""
        disabled={true}
        labels={
          Object {
            "default": "Save",
            "pending": "Saving",
          }
        }
        primary={true}
        state="default"
      />
    </ButtonToolbar>
  </form>
</div>
`;

exports[`StafferForm renders html correctly 1`] = `
<div
  className="create-staffer-form"
>
  <form
    onSubmit={[Function]}
  >
    <Field
      className="staffer-image"
      component={[Function]}
      id="profile_image"
      label={
        <FieldLabel
          className=""
          extraText="Image must be 110x110 pixels in size."
          helpText={
            <div>
              <p>
                Image Requirements:
              </p>
              <ul>
                <li>
                  The image dimensions must be 110×110.
                </li>
                <li>
                  The image size must be less than 256KB.
                </li>
              </ul>
            </div>
          }
          id="image.label"
          optional={false}
          text="Image"
        />
      }
      maxImageSizeKilo={256}
      name="profile_image.medium.url"
      required={false}
      requiredHeight={110}
      requiredWidth={110}
      validate={[Function]}
    />
    <Field
      component={[Function]}
      label={
        <FieldLabel
          className=""
          extraText=""
          helpText=""
          id={null}
          optional={false}
          text="First name"
        />
      }
      name="given_name"
      required={true}
      type="text"
    />
    <Field
      component={[Function]}
      label={
        <FieldLabel
          className=""
          extraText=""
          helpText=""
          id={null}
          optional={false}
          text="Last name"
        />
      }
      name="family_name"
      required={true}
      type="text"
    />
    <Field
      component={[Function]}
      label={
        <FieldLabel
          className=""
          extraText=""
          helpText={
            <div>
              <p>
                Instructor's title at your organization.
              </p>
              <p>
                <b>
                  Examples:
                </b>
              </p>
              <ul>
                <li>
                  Professor
                </li>
                <li>
                  Content Developer
                </li>
                <li>
                  Director
                </li>
              </ul>
            </div>
          }
          id="title.label"
          optional={false}
          text="Title"
        />
      }
      name="position.title"
      required={true}
      type="text"
    />
    <Field
      component={[Function]}
      currentValue=""
      formatCreateLabel={[Function]}
      isCreatable={true}
      label={
        <FieldLabel
          className=""
          extraText=""
          helpText=""
          id={null}
          optional={false}
          text="Organization"
        />
      }
      name="position.organization_override"
      options={Array []}
      placeholder="Organization..."
      validate={[Function]}
    />
    <Field
      component={[Function]}
      id="bio"
      label={
        <FieldLabel
          className=""
          extraText=""
          helpText=""
          id={null}
          optional={false}
          text="Biography"
        />
      }
      maxChars={1000}
      name="bio"
      validate={[Function]}
    />
    <Field
      component={[Function]}
      id="works"
      label={
        <FieldLabel
          className=""
          extraText=""
          helpText=""
          id={null}
          optional={true}
          text="Major works"
        />
      }
      maxChars={250}
      name="major_works"
    />
    <FieldLabel
      className="mb-2"
      extraText=""
      helpText=""
      id={null}
      optional={true}
      text="Social links"
    />
    <FieldArray
      component={[Function]}
      name="urls_detailed"
    />
    <FieldLabel
      className="mb-2"
      extraText=""
      helpText=""
      id={null}
      optional={true}
      text="Areas of expertise"
    />
    <FieldArray
      component={[Function]}
      name="areas_of_expertise"
    />
    <ButtonToolbar
      className=""
      leftJustify={false}
    >
      <Link
        className="btn btn-outline-primary form-cancel-btn"
        disabled={true}
        onClick={[Function]}
        to="/"
      >
        Cancel
      </Link>
      <ActionButton
        className=""
        disabled={true}
        labels={
          Object {
            "default": "Save",
            "pending": "Saving",
          }
        }
        primary={true}
        state="default"
      />
    </ButtonToolbar>
  </form>
</div>
`;

exports[`StafferForm renders html correctly when creating 1`] = `
<div
  className="create-staffer-form"
>
  <form
    onSubmit={[Function]}
  >
    <Field
      className="staffer-image"
      component={[Function]}
      id="profile_image"
      label={
        <FieldLabel
          className=""
          extraText="Image must be 110x110 pixels in size."
          helpText={
            <div>
              <p>
                Image Requirements:
              </p>
              <ul>
                <li>
                  The image dimensions must be 110×110.
                </li>
                <li>
                  The image size must be less than 256KB.
                </li>
              </ul>
            </div>
          }
          id="image.label"
          optional={false}
          text="Image"
        />
      }
      maxImageSizeKilo={256}
      name="profile_image.medium.url"
      required={false}
      requiredHeight={110}
      requiredWidth={110}
      validate={[Function]}
    />
    <Field
      component={[Function]}
      label={
        <FieldLabel
          className=""
          extraText=""
          helpText=""
          id={null}
          optional={false}
          text="First name"
        />
      }
      name="given_name"
      required={true}
      type="text"
    />
    <Field
      component={[Function]}
      label={
        <FieldLabel
          className=""
          extraText=""
          helpText=""
          id={null}
          optional={false}
          text="Last name"
        />
      }
      name="family_name"
      required={true}
      type="text"
    />
    <Field
      component={[Function]}
      label={
        <FieldLabel
          className=""
          extraText=""
          helpText={
            <div>
              <p>
                Instructor's title at your organization.
              </p>
              <p>
                <b>
                  Examples:
                </b>
              </p>
              <ul>
                <li>
                  Professor
                </li>
                <li>
                  Content Developer
                </li>
                <li>
                  Director
                </li>
              </ul>
            </div>
          }
          id="title.label"
          optional={false}
          text="Title"
        />
      }
      name="position.title"
      required={true}
      type="text"
    />
    <Field
      component={[Function]}
      currentValue=""
      formatCreateLabel={[Function]}
      isCreatable={true}
      label={
        <FieldLabel
          className=""
          extraText=""
          helpText=""
          id={null}
          optional={false}
          text="Organization"
        />
      }
      name="position.organization_override"
      options={Array []}
      placeholder="Organization..."
      validate={[Function]}
    />
    <Field
      component={[Function]}
      id="bio"
      label={
        <FieldLabel
          className=""
          extraText=""
          helpText=""
          id={null}
          optional={false}
          text="Biography"
        />
      }
      maxChars={1000}
      name="bio"
      validate={[Function]}
    />
    <Field
      component={[Function]}
      id="works"
      label={
        <FieldLabel
          className=""
          extraText=""
          helpText=""
          id={null}
          optional={true}
          text="Major works"
        />
      }
      maxChars={250}
      name="major_works"
    />
    <FieldLabel
      className="mb-2"
      extraText=""
      helpText=""
      id={null}
      optional={true}
      text="Social links"
    />
    <FieldArray
      component={[Function]}
      name="urls_detailed"
    />
    <FieldLabel
      className="mb-2"
      extraText=""
      helpText=""
      id={null}
      optional={true}
      text="Areas of expertise"
    />
    <FieldArray
      component={[Function]}
      name="areas_of_expertise"
    />
    <ButtonToolbar
      className=""
      leftJustify={false}
    >
      <Link
        className="btn btn-outline-primary form-cancel-btn"
        disabled={true}
        onClick={[Function]}
        to="/"
      >
        Cancel
      </Link>
      <ActionButton
        className=""
        disabled={true}
        labels={
          Object {
            "default": "Save",
            "pending": "Saving",
          }
        }
        primary={true}
        state="pending"
      />
    </ButtonToolbar>
  </form>
</div>
`;

exports[`StafferForm renders html correctly when submitting 1`] = `
<div
  className="create-staffer-form"
>
  <form
    onSubmit={[Function]}
  >
    <Field
      className="staffer-image"
      component={[Function]}
      id="profile_image"
      label={
        <FieldLabel
          className=""
          extraText="Image must be 110x110 pixels in size."
          helpText={
            <div>
              <p>
                Image Requirements:
              </p>
              <ul>
                <li>
                  The image dimensions must be 110×110.
                </li>
                <li>
                  The image size must be less than 256KB.
                </li>
              </ul>
            </div>
          }
          id="image.label"
          optional={false}
          text="Image"
        />
      }
      maxImageSizeKilo={256}
      name="profile_image.medium.url"
      required={false}
      requiredHeight={110}
      requiredWidth={110}
      validate={[Function]}
    />
    <Field
      component={[Function]}
      label={
        <FieldLabel
          className=""
          extraText=""
          helpText=""
          id={null}
          optional={false}
          text="First name"
        />
      }
      name="given_name"
      required={true}
      type="text"
    />
    <Field
      component={[Function]}
      label={
        <FieldLabel
          className=""
          extraText=""
          helpText=""
          id={null}
          optional={false}
          text="Last name"
        />
      }
      name="family_name"
      required={true}
      type="text"
    />
    <Field
      component={[Function]}
      label={
        <FieldLabel
          className=""
          extraText=""
          helpText={
            <div>
              <p>
                Instructor's title at your organization.
              </p>
              <p>
                <b>
                  Examples:
                </b>
              </p>
              <ul>
                <li>
                  Professor
                </li>
                <li>
                  Content Developer
                </li>
                <li>
                  Director
                </li>
              </ul>
            </div>
          }
          id="title.label"
          optional={false}
          text="Title"
        />
      }
      name="position.title"
      required={true}
      type="text"
    />
    <Field
      component={[Function]}
      currentValue=""
      formatCreateLabel={[Function]}
      isCreatable={true}
      label={
        <FieldLabel
          className=""
          extraText=""
          helpText=""
          id={null}
          optional={false}
          text="Organization"
        />
      }
      name="position.organization_override"
      options={Array []}
      placeholder="Organization..."
      validate={[Function]}
    />
    <Field
      component={[Function]}
      id="bio"
      label={
        <FieldLabel
          className=""
          extraText=""
          helpText=""
          id={null}
          optional={false}
          text="Biography"
        />
      }
      maxChars={1000}
      name="bio"
      validate={[Function]}
    />
    <Field
      component={[Function]}
      id="works"
      label={
        <FieldLabel
          className=""
          extraText=""
          helpText=""
          id={null}
          optional={true}
          text="Major works"
        />
      }
      maxChars={250}
      name="major_works"
    />
    <FieldLabel
      className="mb-2"
      extraText=""
      helpText=""
      id={null}
      optional={true}
      text="Social links"
    />
    <FieldArray
      component={[Function]}
      name="urls_detailed"
    />
    <FieldLabel
      className="mb-2"
      extraText=""
      helpText=""
      id={null}
      optional={true}
      text="Areas of expertise"
    />
    <FieldArray
      component={[Function]}
      name="areas_of_expertise"
    />
    <ButtonToolbar
      className=""
      leftJustify={false}
    >
      <Link
        className="btn btn-outline-primary form-cancel-btn"
        disabled={false}
        onClick={[Function]}
        to="/"
      >
        Cancel
      </Link>
      <ActionButton
        className=""
        disabled={false}
        labels={
          Object {
            "default": "Save",
            "pending": "Saving",
          }
        }
        primary={true}
        state="default"
      />
    </ButtonToolbar>
  </form>
</div>
`;
